### PR TITLE
fix: don't touch cache file when use sudo

### PR DESCRIPTION
**Pull request by dead-horse**

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -327,3 +327,9 @@ exports.matchPlatform = function(current, osNames) {
   }
   return hasAnti ? true : false;
 };
+
+exports.isSudo = function() {
```
Comment (dead-horse) [inline lib/utils.js]: hard to write unit test case

Comment (ibigbug) [inline lib/utils.js]: consider use `process.getuid === 0` to detect is root/sudo ?

Comment (dead-horse) [inline lib/utils.js]: 本来是 root 用户没关系，只有通过 sudo 切换到 root 的才需要限制吧

Comment (ibigbug) [inline lib/utils.js]: 原来如此。

File: lib/install.js [hunk 2]
```diff
@@ -47,7 +47,7 @@ function* install(parentDir, pkg, options) {
 function* _install(parentDir, pkg, options) {
   // default install latest version
   if (!pkg.version) {
-    pkg.version = '*';
+    pkg.version = 'latest';
```
Comment (dead-horse) [inline lib/install.js]: - http://registry.npmjs.org/cordova/*
- http://registry.npmjs.org/cordova/latest

npm registry 的处理 * 不等价于 Latest tag

Comment (dead-horse) [inline lib/install.js]: https://github.com/cnpm/cnpmjs.org/commit/b91b160b66c66c924631d9ee5ef588fd5990770c

Comment (dead-horse) [inline lib/install.js]: https://github.com/npm/npm/issues/12804

Comment (dead-horse) [inline lib/install.js]: image

we can't fetch scope package with tag from npm's registry. WTF.

Comment (dead-horse) [inline lib/install.js]: 这里先不管这个问题，cnpm 的处理是一致的，反正在 npm 上也用不了

**Comment by dead-horse** [comment]
感觉 ci 被 force 提交玩坏了，先合并了

**Pull request merged**